### PR TITLE
Add SSO management

### PR DESCRIPTION
**Pull request by itaihanski**
## Related Issues

https://github.com/descope/etc/issues/5810

## Description

Add SSO management

## Must

- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: Descope/Types/Types.cs [hunk 1]
```diff
@@ -674,4 +678,193 @@ public class PasswordSettings
...
+        {
+            Groups = groups;
+            Role = role;
+        }
+    }
+
+    public class SsoSamlSettings
+    {
+        [JsonPropertyName("idpURL")]
+        public string IdpUrl { get; set; }
```
Comment (shilgapira) [inline Descope/Types/Types.cs:696]: Looking at the backend code, this isn't the correct JSON field name. If this was copied from some other SDK you might wanna check if there's a mistake in the original source code as well
```suggestion
        [JsonPropertyName("idpUrl")]
        public string IdpUrl { get; set; }
```

File: Descope/Types/Types.cs [hunk 2]
```diff
@@ -674,4 +678,193 @@ public class PasswordSettings
...
+        [JsonPropertyName("idpEntityId")]
+        public string IdpEntityId { get; set; }
+        [JsonPropertyName("idpCert")]
+        public string IdpCertificate { get; set; }
+        [JsonPropertyName("idpMetadataUrl")]
```
Comment (shilgapira) [inline Descope/Types/Types.cs:701]:
```suggestion
        [JsonPropertyName("attributeMapping")]

```

File: Descope/Types/Types.cs [hunk 3]
```diff
@@ -674,4 +678,193 @@ public class PasswordSettings
         [JsonPropertyName("lockAttempts")]
         public int LockAttempts { get; set; }
     }
+
+    public class RoleMapping
```
Comment (shilgapira) [inline Descope/Types/Types.cs:682]: Do the properties here need `JsonPropertyName` annotations?

Comment (itaihanski) [reply Descope/Types/Types.cs:682]: No - they're manually manipulated

File: Descope/Internal/Management/Sso.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        public async Task ConfigureSAMLSettings(string tenantId, SsoSamlSettings settings, string? redirectUrl = null, List<string>? domains = null)
+        {
+            Utils.EnforceRequiredArgs(
+                ("tenantId", tenantId),
+                ("IdpUrl", settings.IdpUrl),
+                ("IdpCertificate", settings.IdpCertificate),
+                ("IdpEntityId", settings.IdpEntityId));
+
```
[the remark this answers is not in the document]
Comment (itaihanski) [reply Descope/Internal/Management/Sso.cs:29]: I'm good with that

**Review by shilgapira — APPROVED** [review]

**Pull request merged**